### PR TITLE
List table: Improve aria-label and data attributes

### LINKED ISSUE #129: Accessibility Team Review
*Creating this thread for the WP a11y team to drop any issues we find in here and keep them organized*

### PR DESCRIPTION
**Pull request by swissspidy**
See #129

### CODE REVIEW

File: src/class-shiny-updates-list-table.php [hunk 1]
```diff
@@ -439,43 +434,74 @@ public function column_title_translations() {
...
+			case 'theme':
+				$attributes['data-slug'] = esc_attr( $item['slug'] );
+
+				if ( 'button' === $context ) {
+					$attributes['aria-label'] = esc_attr( sprintf( __( 'Update %s now' ), $item['data']->display( 'Name' ) ) );
+				}
+				break;
+			case 'translations':
+				if ( 'button' === $context ) {
+					$attributes['aria-label'] = esc_attr( __( 'Update translations now' ) );
```
Comment (ocean90) [inline src/class-shiny-updates-list-table.php]: `sc_attr( __()` => `esc_attr__()`

File: src/class-shiny-updates-list-table.php [hunk 2]
```diff
@@ -439,43 +434,74 @@ public function column_title_translations() {
...
+				}
+				break;
+			case 'core':
+				$attributes['data-version'] = esc_attr( $item['data']->current );
+				$attributes['data-locale']  = esc_attr( $item['data']->locale );
+
+				if ( 'button' === $context ) {
+					$attributes['aria-label'] = esc_attr( __( 'Update WordPress now' ) );
```
Comment (ocean90) [inline src/class-shiny-updates-list-table.php]: `sc_attr( __()` => `esc_attr__()`

File: src/class-shiny-updates-list-table.php [hunk 3]
```diff
@@ -439,43 +434,74 @@ public function column_title_translations() {
...
+		$attributes = array( 'data-type' => esc_attr( $item['type'] ) );
+
+		switch ( $item['type'] ) {
+			case 'plugin':
+				$attributes['data-plugin'] = esc_attr( $item['slug'] );
+				$attributes['data-slug']   = esc_attr( $item['data']->update->slug );
+				$attributes['data-name']   = esc_attr( $item['data']->Name );
+
+				if ( 'button' === $context ) {
+					$attributes['aria-label'] = esc_attr( sprintf( __( 'Update %s now' ), $item['data']->Name ) );
```
Comment (ocean90) [inline src/class-shiny-updates-list-table.php]: Needs a translator comment.

File: src/class-shiny-updates-list-table.php [hunk 4]
```diff
@@ -439,43 +434,74 @@ public function column_title_translations() {
...
+				}
+				break;
+			case 'theme':
+				$attributes['data-slug'] = esc_attr( $item['slug'] );
+
+				if ( 'button' === $context ) {
+					$attributes['aria-label'] = esc_attr( sprintf( __( 'Update %s now' ), $item['data']->display( 'Name' ) ) );
```
Comment (ocean90) [inline src/class-shiny-updates-list-table.php]: Needs a translator comment.

**Comment by swissspidy** [comment]
Ugh, there are lots of missing translator comments in other places of the code, like the JS templates. I will add these in a separate PR.

**Pull request merged**